app/javascript/components/TagField/index.js

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect, useRef } from 'react'
+
+import debounce from 'lodash.debounce'
+import R from 'ramda'
+
+import '@zendeskgarden/react-tags/dist/styles.css'
+import { Tag, Close } from '@zendeskgarden/react-tags'
+
+import '@zendeskgarden/react-dropdowns/dist/styles.css'
+import { Dropdown, Menu, Item, Multiselect, Field as GardenField } from '@zendeskgarden/react-dropdowns'
+
+const TagField = ({ tags, input: { value, onChange }, maxTags }) => {
+  // TODO: allow admin to set maximum number of tags
+  const [selectedItems, setSelectedItems] = useState(value ? value.map(tag => tag.name) : [])
+  const [inputValue, setInputValue] = React.useState('')
+  const [matchingOptions, setMatchingOptions] = React.useState(tags.map(tag => tag.name))
+
+  /**
+   * Debounce filtering
+   */
+  const filterMatchingOptionsRef = React.useRef(
+    debounce(value => {
+      const matchingOptions = tags
+        .map(tag => tag.name)
+        .filter(option => {
+          return (
+            option
+              .trim()
+              .toLowerCase()
+              .indexOf(value.trim().toLowerCase()) !== -1
+          )
+        })
+
+      setMatchingOptions(matchingOptions)
+    }, 300)
+  )
+
+  useEffect(() => {
+    // Handles existing value and transforms to mutatable format
+    if (value) {
+      onChange(value.map(tag => ({ id: tag.id })))
+    }
+  }, [])
```

Comment:
```suggestion
  }, [value])
```

Reply:
I thought of using `[]` so it loads the graphQL query (loaded once on component mount) into the redux form format. The `onChange` methods write to the redux form in the correct format so I guess it probably doesn't need to call this effect when the `value` changes.